IanGRap/Squish
# Bugfix random deaths

Contributors: Austin

Goal:
Fix the bug where players sometimes die when they dash into blocks

Note:
- Through some debug logging, I found that when the player randomly dies, it is triggered in SquishedFromAbove, which means the top and bottom colliders are picking up 2 _different_ blocks for at least one frame. This means that it happens when a player dashes directly between 2 blocks and Unity physics thinks it is inside the block for a frame.

What changed:
- Reduced the size of player squish detectors so that it is less likely that Unity physics clips them into a block for a frame
- Added a check inside SquishedFromAbove():
    - The block touching the bottom collider must be not moving downwards OR the top block IsPushed

How to Test:
- Load up the match scene and try to get squished by dashing between 2 touching blocks
- Make sure that you still get squished when you should

Note:
- I'm tight on time, so I tested for about 15 minutes and saw no issues. Let me know if you guys still get random squishes.
